lib/pbench/cli/server/shell.py

```diff
@@ -83,21 +83,27 @@ def keycloak_connection(
     # which defaults to 120.0s More detailed documentation on Retry and
     # backoff_factor can be found at:
     # https://urllib3.readthedocs.io/en/latest/reference/urllib3.util.html#module-urllib3.util.retry
-    try:
-        retry = Retry(total=8, backoff_factor=2)
-        adapter = HTTPAdapter(max_retries=retry)
-        session.mount("http://", adapter)
-        response = session.get(f"{oidc_server}/health")
+    retry = Retry(total=8, backoff_factor=2)
+    adapter = HTTPAdapter(max_retries=retry)
+    session.mount("http://", adapter)
+
+    # We will also need to retry the connection if the health status is not UP
+    for i in range(5):
+        try:
+            response = session.get(f"{oidc_server}/health")
+        except Exception:
+            logger.exception("Error connecting to the OIDC client")
+            return return_val + i
         if response.status_code == HTTPStatus.OK and response.json()["status"] == "UP":
-            logger.info("Keycloak connection successful")
-            return
-    except Exception:
-        logger.exception("Error connecting to the OIDC client")
-    else:
-        logger.error(
-            "OIDC client not running, OIDC server response: {}", response.json()
-        )
-    sys.exit(1)
+            logger.debug("OIDC server connection verified")
+            return 0
+        else:
+            return_val += i
+            logger.error(
+                "OIDC client not running, OIDC server response: {}", response.json()
+            )
+            time.sleep(10)
```

Comment:
Consider using [`retry.sleep()`](https://urllib3.readthedocs.io/en/latest/reference/urllib3.util.html#urllib3.util.Retry.sleep) instead.  Or, at least, use the value from [`retry.get_backoff_time()`](https://urllib3.readthedocs.io/en/latest/reference/urllib3.util.html#urllib3.util.Retry.get_backoff_time) instead of a constant `10`.